### PR TITLE
feat: add prettier

### PR DESCRIPTION
**Pull request by noahamac**
Good lessons from this. Never waiting until last minute to run prettier over an entire application :D 

Setting up the github CI is life changing.

### CODE REVIEW

**Comment by noahamac** [comment]
Blocked by #47

File: .eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...nge":"961","text":"878"},{"range":"961","text":"879"},{"range":"962","text":"878"},{"range":"962","text":"879"},{"range":"963","text":"878"},{"range":"963","text":"879"},[2126,2129],"unknown","never",[1970,1973],[1795,1798],[1835,1838],[2077,2080],[2564,2567],[2924,2927],[3391,3394],[1369,1372],[1516,1519],[1897,1900],[30,33],[27,30],[30,33],[27,30],[1798,1801],[2375,2378],[1661,1664],[1949,1952],[3953,3956],[4009,4012],[4081,4084],[4572,4575],[4628,4631],[4700,4703],[5798,5801],[1166,1169],[1186,1189],[1205,1208],[1237,1240],[1355,1358],[3683,3686],[2473,2476],[2591,2594],[4497,4500],[1524,1527],[1205,1208],[1705,1708],[2375,2378],[1148,1151],[1618,1621],[5197,5200],[5228,5231],[5236,5239],[5792,5795],[8120,8123],[8154,8157],[9896,9899],[9930,9933],[10724,10727],[10758,10761],[11529,11532],[11563,11566],[1482,1485],[1661,1664],[1724,1727],[7749,7752],[1198,1201],[1217,1220],[1224,1227],[1314,1317],[1355,1358],[1855,1858],[1867,1870],[1878,1881],[2028,2031],[4084,4087],[4617,4620],[4758,4761],[5360,5363],[5950,5953],[6613,6616],[7089,7092],[7150,7153],[7915,7918],[7962,7965],[8021,8024],[8250,8253],[8536,8539],[8567,8570],[8594,8597],[8625,8628],[8673,8676],[8706,8709],[2099,2102]]
```
Comment (bryans99) [inline .eslintcache:1]: I don't think we want to add this to github

File: .eslintrc.js [hunk 2]
```diff
@@ -23,6 +23,11 @@ module.exports = {
     "@typescript-eslint/explicit-function-return-type": "off",
     "react/prop-types": "off",
     "camelcase": "off",
-    "@typescript-eslint/camelcase": "off"
+    "@typescript-eslint/camelcase": "off",
```
Comment (bryans99) [inline .eslintrc.js:26]: Where are these rules coming from? 

Ultimately we want to unify with sdk codegen and components

File: src/components/DiagramFrame/DiagramCanvas/Diagram.tsx [hunk 3]
```diff
@@ -24,33 +24,33 @@
...
-import * as d3 from 'd3';
-import { useD3 } from '../../../d3-utils/useD3'
-import { addFilter } from '../../../d3-utils/styles'
-import { addZoom } from '../../../d3-utils/zoom'
-import { createLookmlViewElement } from '../../../d3-utils/tables'
-import { createLookmlJoinElement } from '../../../d3-utils/joins'
-import { DiagramJoin } from '../../../utils/LookmlDiagrammer'
-import {DiagramProps} from "./types"
-import {DiagramSpace} from "./components/DiagramSpace"
+import React from "react"
```
Comment (bryans99) [inline src/components/DiagramFrame/DiagramCanvas/Diagram.tsx:27]: Single quotes!

**Review by bryans99 — CHANGES_REQUESTED** [review]
Got as far as noticing that you are using double quotes for imports.

You need to get as close to sdk-codegen as possible.

File: .eslintcache (hunk 1 above)
Comment (noahamac) [reply .eslintcache]: Added to gitignore

File: .eslintrc.js (hunk 2 above)
Comment (noahamac) [reply .eslintrc.js:51]: Brought in the rules from sdk-codegen.

I couldn't figure out why these don't have a definition file defined:
"array-callback-return": "off",
"node/no-path-concat": "off",
"node/no-new-require": "off",
"node/handle-callback-err": "off",

So I disabled them.

And rules like this are to get around d3:
"@typescript-eslint/unbound-method": "off",
"@typescript-eslint/no-empty-function": "warn",

There is work to be done to get 1:1 parity, but this is pretty close now

File: package.json [hunk 4]
```diff
@@ -79,6 +85,12 @@
     "webpack-cli": "^4.6.0",
     "webpack-dev-server": "^3.11.2"
   },
+  "eslintIgnore": [
+    "src/__tests__/**",
```
Comment (bryans99) [inline package.json:89]: Why ignore?

**Review by bryans99 — APPROVED** [review]
This is huge but I'm gonna assume they are mostly automated lint fixes

File: package.json (hunk 4 above)
Comment (noahamac) [reply package.json:89]: still WIP. will lint the spec files

**Review by bryans99 — APPROVED** [review]
Too big. Assuming all lint fixes

**Pull request merged**